BinaryEncoding.md

```diff
@@ -263,8 +249,6 @@ a `data_segment` is:
 
 ### Name section
 
-ID: `name`
-
 The names section does not change execution semantics and a validation error in
 this section does not cause validation for the whole module to fail and is
 instead treated as if the section was absent. The expectation is that, when a
```

Comment:
One idea we discussed was to have the names section be defined as a user-defined section.  Then all the logic here (about how you can always ignore a names section -- that it doesn't have any semantic effect) could be hoisted and apply to all user-defined sections.